tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/SharePointInfo/Search.cs

```diff
@@ -928,10 +928,10 @@ internal bool UnPinThisRecord(ClientContext clientContext, Client client, object
                             Dictionary<string, MatterData> userpinnedMatterCollection = JsonConvert.DeserializeObject<Dictionary<string, MatterData>>(userPinnedMatter);
 
                             if (!string.IsNullOrWhiteSpace(userPinnedDetails.UserPinnedMatterData.MatterName) &&
-                                userpinnedMatterCollection.ContainsKey(WebUtility.HtmlEncode(userPinnedDetails.UserPinnedMatterData.MatterName))
+                                userpinnedMatterCollection.ContainsKey(WebUtility.HtmlEncode(userPinnedDetails.UserPinnedMatterData.MatterName.Trim().ToLower()))
                                 ||
                                 !string.IsNullOrWhiteSpace(userPinnedDetails.UserPinnedMatterData.MatterName) &&
-                                userpinnedMatterCollection.ContainsKey(userPinnedDetails.UserPinnedMatterData.MatterName)
+                                userpinnedMatterCollection.ContainsKey(userPinnedDetails.UserPinnedMatterData.MatterName.Trim().ToLower())
```

Comment:
Fixed the comparison logic for pin and unpin functionality. Implemented case insensitive comparison